src/functional/ffront/decorator.py

```diff
@@ -94,6 +94,26 @@ def _filter_closure_vars_by_type(closure_vars: dict[str, Any], *types: type) ->
     return {name: value for name, value in closure_vars.items() if isinstance(value, types)}
 
 
+def _canonicalize_args(node_params, args, kwargs) -> tuple[tuple, dict]:
+    new_args = []
+
+    for param_i, param in enumerate(node_params):
+        if param.id in kwargs:
+            if param.id not in ["out", "domain"] and param_i < len(args):
```

Comment:
For my own understanding why are `out` and `domain` excluded?

Reply:
e.g. when you call: ```program(input_1, b=input_2, c=input_3, out=out, offset_provider={})````

In this case out is also a keyword argument

Reply:
I did some testing, you are right